### PR TITLE
Added dynamic configuration of http port in the unity sdk

### PR DESCRIPTION
**Pull request by roberthbailey**
@dzlier-gcp -- I'm looking for feedback on the way that I'm getting the environment variable and converting it to an integer. This follows the pattern I implemented for [Go](https://github.com/googleforgames/agones/pull/1086) / [nodejs](https://github.com/googleforgames/agones/pull/1093), but [in the C++ sdk](https://github.com/googleforgames/agones/pull/1092) I went with a more concise solution (with less logging / error checking) since that was considered more stylistically correct. 

Part of #851.

### CODE REVIEW

File: sdks/unity/AgonesSdk.cs [hunk 1]
```diff
@@ -131,6 +133,35 @@ public async Task<bool> SetAnnotation(string key, string value)
...
+            catch (FormatException)
+            {
+                // Unable to parse value defined in AGONES_SDK_HTTP_PORT into an integer.
+                Console.WriteLine("Unable to parse value defined in AGONES_SDK_HTTP_PORT into an integer. Using default port {0}.", sidecarPort);
+                return;
+            }
+            if (port < 1 || port > 65535)
+            {
+                // Invalid port defined in AGONES_SDK_HTTP_PORT. It must be between 1 and 65535.
+                Console.WriteLine("Invalid port defined in AGONES_SDK_HTTP_PORT: it must be between 1 and 65535. Using default port {0}", sidecarPort);
```
[the remark this answers is not in the document]
Comment (roberthbailey) [reply sdks/unity/AgonesSdk.cs]: I'm not convinced that it should fail (but feel free to convince me!). My expectation was that if the game server was unable to connect to the sidecar, the first thing you would do is run `kubectl logs` on the game server, see that it was trying to connect to the wrong port, see this message that the port in the environment was invalid, and then fix it. 

Is it better to just see that it's trying to connect to the wrong port (e.g. "unable to connect to http://127.0.0.1:666666"? That would be less code (and would be in line with how the C++ API turned out).

[the remark this answers is not in the document]
Comment (pooneh-m) [reply sdks/unity/AgonesSdk.cs]: I prefer fail fast for easier debugging before the game is on. If the sidecar is up and running, debugging will be redirected to game server and time will be wasted to identify the root cause. Put yourself in the shoe of a developer with a bug in the environment variable. Which one do you prefer to immediately know about your bug or learn about it when things don't work as expected?

File: sdks/unity/AgonesSdk.cs [hunk 2]
```diff
@@ -153,7 +184,7 @@ private async Task<bool> SendRequestAsync(string api, string json, string method
             // To prevent that an async method leaks after destroying this gameObject.
             cancellationTokenSource.Token.ThrowIfCancellationRequested();
 
-            var req = new UnityWebRequest(sidecarAddress + api, method)
+            var req = new UnityWebRequest(sidecarHost + ":" + sidecarPort + api, method)
             {
```
[the remark this answers is not in the document]
Comment (roberthbailey) [reply sdks/unity/AgonesSdk.cs]: I've changed this back to how it looked originally.

File: sdks/unity/AgonesSdk.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (roberthbailey) [reply sdks/unity/AgonesSdk.cs]: I've gone ahead and taken the same approach I ended up with in the C++ SDK: just use the environment variable directly and let the connection libraries throw the error. The code is much shorter/simpler. Let me know what you think.

**Comment by roberthbailey** [comment]
> In fact when this file was added I asked if that port was constant or if it should be a variable, and was assured it was a constant heh.

It was ... until we found a bug and decided it needed to be changed. Since we can't ensure that whatever port we pick will always be free of conflicts, we decided to make it configurable.

File: sdks/unity/AgonesSdk.cs [hunk 3]
```diff
@@ -58,6 +58,8 @@ private struct KeyValueMessage
         // Use this for initialization.
         private void Start()
         {
+            String port = Environment.GetEnvironmentVariable("AGONES_SDK_HTTP_PORT");
+            sidecarAddress = "http://localhost:" + (port != null ? port : "59358");
```
Comment (pooneh-m) [inline sdks/unity/AgonesSdk.cs]: you can replace it with ?? 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator

Comment (roberthbailey) [reply sdks/unity/AgonesSdk.cs]: Sweet. I've updated it.

**Pull request merged**